### PR TITLE
Introduce setenv exec command

### PR DESCRIPTION
**Pull request by potsbo**
## WHY

We want to use accounts information stored in env var.
Therefore, we need `exec` command to run any command as child process

## WHAT

- [x] Add setenv command

### CODE REVIEW

**Comment by potsbo** [comment]
It works
```console
potsbo@magi-INS-% make && envchain wtd bin/developers-account-mapper setenv potsbo exec env
make: `bin/developers-account-mapper' is up to date.
<omitted>
GITHUB_USERNAME=potsbo
```

**Comment by potsbo** [comment]
@dtan4 Implemented, please review this.

**Comment by dtan4** [comment]
How about `developers-account-mapper exec potsbo <command>` ? Are you planning to implement some other subcommand additionaly?

**Comment by potsbo** [comment]
No, I don't have any plan other than `exec`.
I just thought people would think it's normal that commands come right after `exec`

**Comment by potsbo** [comment]
Decided to use exec as a subcommand.
Here is the test.

```console
potsbo@magi-INS-% make && envchain wtd bin/developers-account-mapper exec potsbo env
go build -ldflags="-s -w -X \"main.Version=0.1.0\" -X \"main.Revision=750b1d2\"" -o bin/developers-account-mapper
<omitted>
GITHUB_USERNAME=potsbo
```

@dtan4 review again please

File: commands.go [hunk 1]
```diff
@@ -17,6 +17,11 @@ func Commands(meta *command.Meta) map[string]cli.CommandFactory {
 				Meta: *meta,
 			}, nil
 		},
+		"setenv": func() (cli.Command, error) {
```
Comment (dtan4) [inline commands.go]: 👁

Comment (potsbo) [reply commands.go]: forgot to push...

File: command/exec.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	envs = append(envs, fmt.Sprintf("%s=%s", "GITHUB_USERNAME", user.GitHubUsername))
+
+	execCmd := exec.Command(args[1], args[2:]...)
+	execCmd.Env = envs
+	execCmd.Stderr = os.Stderr
+	execCmd.Stdout = os.Stdout
+	err = execCmd.Run()
+	if err != nil {
+		log.Println(err)
+		return 1
```
Comment (dtan4) [inline command/exec.go:50]: Sometimes command returns exit code besides 1 at failure time, so `exec` subcommand should return the value as it is.

ref: https://github.com/dtan4/valec/blob/cc6fac0f599aeec8f7d3d81fd6d5f054a09f247b/cmd/exec.go#L47-L57

Comment (potsbo) [reply command/exec.go:50]: @dtan4 
Implemented as instructed, but I can not fetch subcommand exit status.

```console
potsbo@magi-INS-% envchain wtd bin/developers-account-mapper exec potsbo foo
2017/01/11 10:45:38 exec: "foo": executable file not found in $PATH
potsbo@magi-INS-% echo $?
1
potsbo@magi-INS-% foo
zsh: command not found: foo
potsbo@magi-INS-% echo $?
127
```

Before I look into the detail of go and system call, I want your opinion, am I doing anything wrong or this is how it is?

Comment (dtan4) [reply command/exec.go:50]: It works with _existed_ command:

```bash
$ cat ./127.sh
#!/bin/bash

exit 127
```

```bash
$ ./127.sh
$ echo $?
127
$ envchain wtd bin/developers-account-mapper exec potsbo ./127.sh
$ echo $?
127
```

Your reply shows that `exec` command showed error of `execCmd.Run()`and returned 1 for subcommand _which does not exist_.
https://github.com/wantedly/developers-account-mapper/blob/aaa7207318c4c8124bd24275b0a07f28790b5572/command/exec.go#L48-L51

Comment (potsbo) [reply command/exec.go:50]: It's a very kind follow-up. I understand, thank you.
Then, this is ok to merge?

Comment (dtan4) [reply command/exec.go:50]: yeah, LGTM 👍

**Review by dtan4 — APPROVED** [review]

**Comment by dtan4** [comment]
LGTM 👍

**Pull request merged**